src/templates/PoolV2/Activity/index.tsx

```diff
@@ -143,29 +144,50 @@ const Activity = () => {
 
     return activities.sort((a, b) => b.date.getTime() - a.date.getTime())
   }, [data, optionsSelected])
+
   return (
     <S.Activity>
+      <S.FilterContainer>
+        <S.FilterIconContainer onClick={() => setIsOpenFilter(!isOpenFilter)}>
```

Comment:
When I have to access the previous value of a state inside the `setter`, I tend to use its callback, like.
You should do it, just to make sure you're always working with the latest state. Without using the callback, you can't be sure it's the correct latest state.
```suggestion
        <S.FilterIconContainer onClick={() => setIsOpenFilter(prev => !prev)}>
```